### ISSUE TITLE
[feature request] allow for custom names in meta tags

### ISSUE THREAD

**Issue by Dantali0n** [omeka-s-modules/ResourceMeta#4]
The available fields in the ResourceMeta module are not expanded / driven by the custom vocabularies that are installed. It would be nice if the available fields would be derived from these vocabularies or if we could create custom tags altogether.

**Comment by jimsafley** [comment]
Could you elaborate on this? Im not sure I understand what feature you're requesting.

**Comment by Dantali0n** [comment]
@jimsafley Ofcourse,

Currently, the list available as meta tags in ResourceMeta is fed from a static list in the config file.

While internally Omeka-s already has vocabularies, vocabularies the user can extend, modify, remove etc.
![image](https://user-images.githubusercontent.com/5026295/235674739-18498070-9a6b-4967-b5fd-eda8bfa31222.png)

I think it is a better, more long term solution, if either the user can define any meta tag they want entirely or the available meta tags are fed from the vocabularies.

In Short, I used the CustomOntology module to create my own vocabulary and added this to my Omeka-s instance. Since many of the fields available in ResourceMeta are the same as the default vocabularies I expected this to extend my options but it did not since they are hardcoded in the config.

Hope this makes more sense now.

**Comment by jimsafley** [comment]
Thank you. I'm hesitant to add meta names that are not expressly used by crawlers or clients. The current list of names derives from [guidelines for inclusion into Google Scholar](https://scholar.google.com/intl/en/scholar/inclusion.html#indexing), so there's a defined, bibliographic use for them. Adding a name for every property in an installation would bloat an already extensive list, for what I perceive to be little benefit. I'm open to a feature where administrators add custom names, but I prefer recommendations and pull requests that add names that have demonstrable use (like [your own PR](https://github.com/omeka-s-modules/ResourceMeta/pull/5)).

**Comment by Dantali0n** [comment]
Indeed, perhaps it is not good to pre-populate this list with properties from vocabularies which are never used. However, the amount of meta tags, especially once considering [Open Graph](https://ogp.me/) and [microformats](https://developer.mozilla.org/en-US/docs/Web/HTML/microformats) becomes extensive rather quickly.

Maybe we should populate the list with sane defaults, like you already have, include any new tags if we think they are important enough. But in the end allow to user to define a custom tag, possibly acknowledging a warning or enabling some advanced user mode. This would allow users to still define any tag that we potentially miss without cluttering this already extensive list any further.

**Comment by jimsafley** [comment]
Note that the [Sharing](https://github.com/omeka-s-modules/Sharing) module already implements Open Graph, but this module probably should as well. I agree that custom names is a good compromise, possibly under a reserved "User defined" option group. I'll look into it in the coming months.
